File: Hunt_Career_Playwright/pages/SearchPage.js
import { BasePage } from "./BasePage";

export class SearchPage extends BasePage {
  constructor(page) {
    super(page);
    this.page = page;

    // Elements
    this.searchInput = this.page.locator(
      "//input[@placeholder='Search jobs by title, company, or keyword...']",
    );
    this.searchButton = this.page.locator("//button[@aria-label='Search']");
    this.locationButton = this.page.locator(
      "//button[@aria-label='Toggle location filter dropdown']",
    );
    this.locationSearchInput = this.page.locator(
      "//input[@placeholder='Search location...']",
    );
    this.jobTypeButton = this.page.locator(
      "//button[@aria-label='Toggle job type filter dropdown']",
    );
    this.jobTypeSearchInput = this.page.locator(
      "//input[@placeholder='Search job type...']",
    );
    this.clearAllFiltersButton = this.page.locator(
      "//button[@aria-label='Clear all filters']",
    );
    this.loginSuccessMessage = this.page.locator(
      "//div[contains(text(), 'Logged in successfully!')]",
    );
    this.noJobsFoundMessage = this.page.locator(
      'p:has-text("No jobs found.")',
    );
  }

  // Dynamic locators
  searchTag(searchTerm) {
    return this.page.locator(`//span[contains(., "🔍 ${searchTerm}")]`);
  }

  locationTag(location) {
    return this.page.locator(`//span[contains(., "📍 ${location}")]`);
  }

  jobTypeTag(jobType) {
    return this.page.locator(`//span[contains(., "🧾 ${jobType}")]`);
  }

  locationOption(location) {
    return this.page.locator(`label:has-text("${location}")`);
  }

  jobTypeOption(jobType) {
    return this.page.locator(`label:has-text("${jobType}")`);
  }

  jobCardTitle(index) {
    return this.page
      .locator(
        "//h3[contains(@class, 'text-lg') and contains(@class, 'font-semibold')]",
      )
      .nth(index);
  }

  // Actions
  async navigateTo() {
    await this.navigateToJobSeeker();
  }

  async search(searchTerm, location = null, jobType = null) {
      await this.typeInElement(this.searchInput, searchTerm);
      await this.clickElement(this.searchButton);

    if (location) {
      await this.clickElement(this.locationButton);
      await this.typeInElement(this.locationSearchInput, location);
      await this.clickElement(this.locationOption(location));
      await this.clickElement(this.locationButton);
    }

    if (jobType) {
      await this.clickElement(this.jobTypeButton);
      await this.clickElement(this.jobTypeOption(jobType));
      await this.clickElement(this.jobTypeButton);
    }
  }
}

File: Hunt_Career_TestNG_Hybrid_Framework/src/main/java/com/huntcareer/qa/pages/SearchPage.java
package com.huntcareer.qa.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class SearchPage extends PageBase {

    @FindBy(xpath = "//input[@placeholder='Search jobs by title, company, or keyword...']")
    private WebElement SearchInputField;

    @FindBy(xpath = "//button[@aria-label='Search']")
    private WebElement SearchButton;

    @FindBy(xpath = "//button[@aria-label='Toggle location filter dropdown']")
    private WebElement LocationButton;

    @FindBy(xpath = "//input[@placeholder='Search location...']")
    private WebElement LocationSearchInput;

    @FindBy(xpath = "//button[@aria-label='Toggle job type filter dropdown']")
    private WebElement JobTypeButton;

    @FindBy(xpath = "//input[@placeholder='Search job type...']")
    private WebElement JobTypeSearchInput;

    @FindBy(xpath = "//button[@aria-label='Clear all filters']")
    private WebElement ClearAllFiltersButton;

    @FindBy(xpath = "//p[contains(text(),'No jobs found.')]")
    public WebElement noJobsFoundMessage;

    public SearchPage(WebDriver driver) {
        super(driver);
        PageFactory.initElements(driver, this);
    }

    public void enterKeywordIntoSearchInputField(String searchTerm) {
        typeInElement(SearchInputField, searchTerm);
    }

    public void clickOnSearch() {
        clickElement(SearchButton);
    }

    public void clickLocationButton() {
        clickElement(LocationButton);
    }

    public void enterLocationSearchInput(String locationText) {
        typeInElement(LocationSearchInput, locationText);
    }

    public void selectLocationOption(String location) {
        clickElement(driver.findElement(By.xpath("//label[contains(., '" + location + "')]/input")));
    }

    public void clickJobTypeButton() {
        clickElement(JobTypeButton);
    }

    public void enterJobTypeSearchInput(String jobTypeText) {
        typeInElement(JobTypeSearchInput, jobTypeText);
    }

    public void selectJobTypeOption(String jobType) {
        clickElement(driver.findElement(By.xpath("//label[contains(., '" + jobType + "')]/input")));
    }

    public void clickClearFilterOption() {
        clickElement(ClearAllFiltersButton);
    }

    public WebElement getJobCardTitle(WebElement jobcard) {
        return jobcard.findElement(By.cssSelector("h3.text-lg.font-semibold"));
    }

    public WebElement jobCardCompanyLocation(WebElement jobcard) {
        return jobcard.findElement(By.cssSelector("p.text-gray-600"));
    }

    public WebElement jobCardDescription(WebElement jobcard) {
        return jobcard.findElement(By.cssSelector("p.text-sm.text-gray-700.mt-3.line-clamp-3"));
    }

    public WebElement getSearchTag(String searchTerm) {
        return driver.findElement(By.xpath("//span[contains(text(), '" + searchTerm + "')]"));
    }

    public WebElement getLocationTag(String location) {
        return driver.findElement(By.xpath("//span[contains(text(), '" + location + "')]"));
    }

    public WebElement getJobTypeTag(String jobType) {
        return driver.findElement(By.xpath("//span[contains(text(), '" + jobType + "')]"));
    }
}
